packages/compiler-cli/src/ngtsc/annotations/src/component.ts

```diff
@@ -271,20 +299,41 @@ export class ComponentDecoratorHandler implements
     let styles: string[]|null = null;
 
     const styleResources = this._extractStyleResources(component, containingFile);
-    const styleUrls = this._extractStyleUrls(component, template.styleUrls);
-    if (styleUrls !== null) {
-      if (styles === null) {
-        styles = [];
+    let styleUrls: StyleUrl[]|null = null;
+
+    const componentStyles = this._extractComponentStyleUrls(component);
+    const templateStyles = this._extractTemplateStyleUrls(template);
+
+    if (componentStyles !== null) {
+      if (styleUrls === null) {
+        styleUrls = [];
+      }
+      styleUrls.push(...componentStyles);
+    }
```

Comment:
```suggestion
    const componentStyles = this._extractComponentStyleUrls(component);
    const templateStyles = this._extractTemplateStyleUrls(template);

    const styles: string[] = [];
    const styleUrls: StyleUrl[] = [...componentStyles, ...templateStyles];
```

It looks like there's not really a reason `styles` or `styleUrls` should be set to `null`. In the end, the output for `styles` is `styles: styles || []` and `styleUrls` is just iterated so it makes no difference if it's empty or `null`.
In addition to that, there would then not really be a reason for `_extractComponentStyleUrls` or `_extractTemplateStyleUrls` to return `null` either but can instead return empty array.

Reply:
I think the existing code is trying to avoid allocating an empty array when it's not necessary for better perf.

For `styles`, initializing as an empty array would not cause a performance hit because the output ends up as `styles: styles || []`. Let's go ahead and initialize it as `[]` ✔️ 

`styleUrls` is null in the output when there are no styles, so changing it to empty array would *theoretically* cause a perf hit. Initializing it as `[]` would cause 1 extra array to be allocated each analysis phase for components with no styles. If you ask me, I'd say this is probably fine, but when do we need to be careful about this?

Reply:
I would really not think this theoretical performance difference is significant enough to matter.

Reply:
okay, I'll change it to use `[]` instead of `null` ✔️